funnel/models/space.py

```diff
@@ -109,24 +120,26 @@ def proposals_all(self):
             return self.proposals
 
     @property
-    def proposals_by_status(self):
-        from .proposal import Proposal, PROPOSALSTATUS
+    def proposals_by_state(self):
+        from .proposal import Proposal
         if self.subspaces:
             basequery = Proposal.query.filter(Proposal.proposal_space_id.in_([self.id] + [s.id for s in self.subspaces]))
         else:
             basequery = Proposal.query.filter_by(proposal_space=self)
-        return dict((status, basequery.filter_by(status=status).order_by(db.desc('created_at')).all()) for (status, title) in PROPOSALSTATUS.items() if status != PROPOSALSTATUS.DRAFT)
+        all_proposals = basequery.filter(~Proposal.state.DRAFT).order_by(db.desc('created_at'))
+        all_by_state = Proposal.state.group(all_proposals)
+        return all_by_state
 
     @property
     def proposals_by_confirmation(self):
-        from .proposal import Proposal, PROPOSALSTATUS
+        from .proposal import Proposal
         if self.subspaces:
             basequery = Proposal.query.filter(Proposal.proposal_space_id.in_([self.id] + [s.id for s in self.subspaces]))
         else:
             basequery = Proposal.query.filter_by(proposal_space=self)
         response = dict(
```

Comment:
`response` is a slightly confusing variable name in this context. Like above you don't need this variable, you can return `dict(...)` directly.